src/main/java/uk/gov/companieshouse/api/strikeoffobjections/file/FileTransferApiClient.java

```diff
@@ -0,0 +1,133 @@
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Component;
+import org.springframework.util.LinkedMultiValueMap;
+import org.springframework.web.client.RestTemplate;
+import org.springframework.web.multipart.MultipartFile;
+import uk.gov.companieshouse.api.strikeoffobjections.common.ApiLogger;
+
+import java.io.IOException;
+
+@Component
+public class FileTransferApiClient {
+
+    private static final String HEADER_API_KEY = "x-api-key";
+    private static final String UPLOAD = "upload";
+    private static final String CONTENT_DISPOSITION_VALUE = "form-data; name=%s; filename=%s";
+    private static final String NULL_RESPONSE_MESSAGE = "null response from file transfer api url";
+
+    @Autowired
+    private ApiLogger logger;
+
+    @Autowired
+    private RestTemplate restTemplate;
+
+    @Value("${FILE_TRANSFER_API_URL}")
+    private String fileTransferApiURL;
+
+    @Value("${FILE_TRANSFER_API_KEY}")
+    private String fileTransferApiKey;
+
+
+    private <T> FileTransferApiClientResponse makeApiCall(String requestId,
+                                                          FileTransferOperation <T> operation,
+                                                          FileTransferResponseBuilder <T> responseBuilder) {
+        FileTransferApiClientResponse response = new FileTransferApiClientResponse();
+
+        try {
+            T operationResponse = operation.execute();
+            response = responseBuilder.createResponse(operationResponse);
+        } catch (IOException e) {
+            logger.errorContext(requestId, e.getMessage(), e);
+            response.setHttpStatus(HttpStatus.INTERNAL_SERVER_ERROR);
+        }
+        return response;
+    }
+
+    /**
+     * Uploads a file to the file-transfer-api
+     * Creates a multipart form request containing the file and sends to
+     * the file-transfer-api. The response from the file-transfer-api contains
+     * the new unique id for the file. This is captured and returned in the FileTransferApiClientResponse.
+     * @param fileToUpload The file to upload
+     * @return FileTransferApiClientResponse containing the file id if successful, the http header and http status
+     */
+    public FileTransferApiClientResponse upload(String requestId, MultipartFile fileToUpload) {
+
+        return makeApiCall(
+             requestId,
+             () -> getFileTransferOperation(fileToUpload),
+             responseEntity ->  getFileTransferApiClientResponse(requestId, responseEntity)
+        );
+    }
+
+    /**
+     * Calls the file transfer api and returns the result of the
+     * call ready to pass to the response builder
+     * @param fileToUpload
+     * @return ResponseEntity containing the raw data from which the response object is built
+     * @throws IOException
```

Comment:
Since you've included the @throws Javadoc tag you should add some words around when that exception might be thrown.

Reply:
Added information.